odoo/addons/test_lint/tests/_odoo_checker_unlink_override.py

```diff
@@ -0,0 +1,44 @@
+import astroid
+from pylint import checkers, interfaces
+
+
+class OdooBaseChecker(checkers.BaseChecker):
+    __implements__ = interfaces.IAstroidChecker
+    name = 'odoo'
+
+    msgs = {
+        'E8503': (
+            'Raise inside unlink override.',
+            'raise-unlink-override',
+            'Raising errors is not allowed inside unlink overrides, '
+            'you can create a method and decorate it with '
+            '@api.ondelete(at_uninstall=False), only use '
+            'at_uninstall=True if you know what you are doing.'
+        )
+    }
+
+    @staticmethod
+    def _will_raise(node):
+        body = getattr(node, 'body', [])
+        if any(isinstance(n, astroid.Raise) for n in body):
+            return True
+        for sub_node in body:
+            res = OdooBaseChecker._will_raise(sub_node)
+            if res:
+                return res
+        return False
+
+    @checkers.utils.check_messages('raise-unlink-override')
+    def visit_classdef(self, node):
```

Comment:
maybe the reverse is easier to do: `visit_raise` and check parents for `astroid.FuncDef` with `name == 'unlink'` and `astroid.ClassDef` with `BaseModel` in ancestors